server/svix-server/src/worker.rs

```diff
@@ -504,47 +504,67 @@ pub async fn worker_loop(
     queue_tx: TaskQueueProducer,
     mut queue_rx: TaskQueueConsumer,
     op_webhook_sender: OperationalWebhookSender,
+    mut term_rx: tokio::sync::watch::Receiver<bool>,
 ) -> Result<()> {
     loop {
-        match queue_rx.receive_all().await {
-            Ok(batch) => {
-                for delivery in batch {
-                    let cfg = cfg.clone();
-                    let pool = pool.clone();
-                    let cache = cache.clone();
-                    let queue_tx = queue_tx.clone();
-                    let queue_task = delivery.task.clone();
-                    let op_webhook_sender = op_webhook_sender.clone();
-
-                    tokio::spawn(async move {
-                        let worker_context = WorkerContext {
-                            cfg: &cfg,
-                            db: &pool,
-                            cache: &cache,
-                            queue_tx: &queue_tx,
-                            op_webhook_sender: &op_webhook_sender,
-                        };
-
-                        if let Err(err) = process_task(worker_context, queue_task).await {
-                            tracing::error!("Error executing task: {}", err);
-                            queue_tx
-                                .nack(delivery)
-                                .await
-                                .expect("Error sending 'nack' to Redis after task execution error");
-                        } else {
-                            queue_tx.ack(delivery).await.expect(
-                                "Error sending 'ack' to Redis after successful task execution",
-                            );
+        tokio::select! {
```

Comment:
Using select will mean that the first to finish "wins", which in turns mean that a signal is not a graceful shutdown, it's violently killing the async task.

It should just be an `if` in a place where we are OK with killing (sort of like a "yield"). So just at the beginning of every loop for example, check the value and see if we are good to stop. We don't even need a watch, it can literally just be a variable that we check. Heck, it can even be a global variable with `lazy_static!` so we don't even need to pollute the code.